controller/context/cachekeycontext/finalizers_kept_context.go

```diff
@@ -0,0 +1,30 @@
+// Package cachekeycontext stores and accesses the local context key in
+// context.Context.
+package cachekeycontext
+
+import (
+	"context"
+	"strconv"
+	"time"
+)
+
+// key is an unexported type for keys defined in this package. This prevents
+// collisions with keys defined in other packages.
+type key string
+
+// ctxKey is the key for kept values in context.Context. Clients use
+// cachekeycontext.NewContext and cachekeycontext.FromContext instead of using
+// this key directly.
+var ctxKey key = "kept"
+
+// NewContext returns a new context.Context that can be used to access the
+// current reconciliation loop's cache key.
+func NewContext(ctx context.Context) context.Context {
+	return context.WithValue(ctx, ctxKey, strconv.Itoa(int(time.Now().UnixNano())))
+}
```

Comment:
The story around this is to be able to implement some caching mechanisms in the operators. I would like to start aiming for self sufficient resource handlers that do not make use of the current controller context concept anymore. This implies that all the information we put into the controller context have to be fetched and accessed regardless. In case resource handlers are self sufficient they leverage certain service implementations which should not just call remote APIs over and over again. These service implementations can make use of the operatorkit cache key added to the context. The cache key is unique per reconciliation loop, given reconciliation loops are executed sequentially. The assumption here is that using a timestamp as poor man's cache key is sufficient.